src/main/java/org/broadinstitute/hellbender/tools/walkers/conversion/GtfToBed.java

```diff
@@ -0,0 +1,288 @@
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.util.*;
+import htsjdk.tribble.Feature;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ShortVariantDiscoveryProgramGroup;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.codecs.gtf.GencodeGtfFeature;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * <p>Convert Gencode GTF files to BED format with options for gene and transcript level processing.
+ * This tool allows for the extraction of gene and transcript information from Gencode GTF files and
+ * outputs the data in BED format.</p>
+ *
+ *
+ * <p>The conversion process includes sorting entries
+ * by karyotype, providing flexibility in the selection of either gene or transcript level
+ * data, and an option to only use basic tags. It ensures that the BED output is sorted and formatted correctly for subsequent use.
+ * Note that it has been tested for both human and mouse Gencode GTFs. </p>
+ *
+ * <h3>Usage examples</h3>
+ * <p>Example commands to run GtfToBed for typical scenarios:</p>
+ *
+ * <h4>(i) Convert GTF to BED with gene level data</h4>
+ * <p>This mode extracts and converts gene data from the input GTF file to BED format:</p>
+ *
+ * <pre>
+ *     java -jar GtfToBed.jar \
+ *     -G input.gtf \
+ *     -SD dictionary.dict \
+ *     -T False \
+ *     -B False \
+ *     -O output.bed \
+ * </pre>
+ *
+ * <h4>(ii) Convert GTF to BED with transcript level data</h4>
+ * <p>This mode extracts and converts transcript data from the input GTF file to BED format:</p>
+ *
+ * <pre>
+ *     java -jar GtfToBed.jar \
+ *     -G input.gtf \
+ *     -SD dictionary.dict \
+ *     -T True \
+ *     -B False \
+ *     -O output.bed \
+ * </pre>
+ *
+ * <h4>(iii) Convert GTF to BED with transcript level data filtering for only those with the basic tag</h4>
+ *  * <p>This mode extracts and converts basic transcript data from the input GTF file to BED format:</p>
+ *  *
+ *  * <pre>
+ *     java -jar GtfToBed.jar \
+ *     -G input.gtf \
+ *     -SD dictionary.dict \
+ *     -T True \
+ *     -B True \
+ *     -O output.bed \
+ *  * </pre>
+ */
+
+@CommandLineProgramProperties(
+        summary = "Converts Gencode GTF files to Bed file format with each row of bed file being either a gene or a transcript.",
+        oneLineSummary = "Gencode GTF to BED",
+        programGroup = ShortVariantDiscoveryProgramGroup.class
+)
+
+public class GtfToBed extends FeatureWalker<GencodeGtfFeature> {
```

Comment:
Don't forget the  `@DocumentedFeature` tag to make it show up in the help docs.  Consider  adding the necessary `@WorkflowProperties` so it will be made into a WDL automatically also.